### DISCUSSION TITLE
基因富集分析

### DISCUSSION THREAD

**Discussion by zhaoliang0302** [ShixiangWang/self-study#45] [Q&A] [answered]
翔哥好，
我在看文献时看到作者描述他对于一个自定义基因（module的基因）集富集分析的描述
>  Functional annotation of modules was carried out in the R statistical computing environment using a one-sided Fisher's Exact Test for overrepresentation of the MSigDB v6·1 gene sets (Hallmark, Canonical Pathways, Gene Ontology) 

这里的fisher检验难道是根据自己的基因集是否落在目标基因集，做四个表检验吗？我觉得如果这样的话是不是不够严谨，因为在go、kegg这种分析中不是这样做的。
但是我不知道自定义基因怎么看是否富集在感兴趣的基因集中，请问这种流程如何实现较为科学呢？
谢谢

**Comment by ShixiangWang** [comment, marked as answer]
可以做的，fisher 本身常用于富集分析。洲更之前写过一篇文章有介绍过，还蛮不错的，这是一个链接 https://www.plob.org/article/11508.html

重点是生成二联表

![image](https://user-images.githubusercontent.com/25057508/82345092-707f1480-9a27-11ea-9822-9b56a734b38a.png)

下面是来自最经典的示例：

```r
> TeaTasting <-
+     matrix(c(3, 1, 1, 3),
+            nrow = 2,
+            dimnames = list(Guess = c("Milk", "Tea"),
+                            Truth = c("Milk", "Tea")))
> fisher.test(TeaTasting, alternative = "greater")

	Fisher's Exact Test for Count Data

data:  TeaTasting
p-value = 0.2429
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 0.3135693       Inf
sample estimates:
odds ratio 
  6.408309 

> TeaTasting
      Truth
Guess  Milk Tea
  Milk    3   1
  Tea     1   3
```

**Comment by zhaoliang0302** [comment]
了解了，谢谢翔哥
